storage/src/main/java/org/apache/kafka/storage/internals/log/UnifiedLog.java

```diff
@@ -1100,7 +1102,8 @@ private LogAppendInfo append(MemoryRecords records,
                                  VerificationGuard verificationGuard,
                                  boolean ignoreRecordSize,
                                  byte toMagic,
-                                 short transactionVersion) {
+                                 short transactionVersion,
+                                 boolean isLeader) {
```

Comment:
We don't need to pass in isLeader. isLeader can be computed as origin != REPLICATION.